src/MergePlugin.php

```diff
@@ -333,14 +317,13 @@ public function onPostInstallOrUpdate(ScriptEvent $event)
         // @codeCoverageIgnoreStart
         if ($this->state->isFirstInstall()) {
             $this->state->setFirstInstall(false);
-            $this->logger->info(
-                '<comment>' .
-                'Running additional update to apply merge settings' .
-                '</comment>'
-            );
+            $this->logger->log("\n".'<info>Running composer update to apply merge settings</info>');
 
-            $config = $this->composer->getConfig();
+            $file = Factory::getComposerFile();
+            $lock = Factory::getLockFile($file);
```

Comment:
This method does not exist in Composer 1. This issue arises when installing the plugin for the first time.

> This PR seems to work mostly correctly.
> However, I am getting this error when installing with Composer 1.10.1:
> ```
> PHP Fatal error:  Uncaught Error: Call to undefined method Composer\Factory::getLockFile() in /mnt/data/workspace/drupal_d9/vendor/wikimedia/composer-merge-plugin/src/MergePlugin.php:323
> Stack trace:
> #0 [internal function]: Wikimedia\Composer\MergePlugin->onPostInstallOrUpdate()
> #1 /usr/share/php/Composer/EventDispatcher/EventDispatcher.php(164): call_user_func()
> #2 /usr/share/php/Composer/EventDispatcher/EventDispatcher.php(96): Composer\EventDispatcher\EventDispatcher->doDispatch()
> #3 /usr/share/php/Composer/Installer.php(338): Composer\EventDispatcher\EventDispatcher->dispatchScript()
> #4 /usr/share/php/Composer/Command/InstallCommand.php(122): Composer\Installer->run()
> #5 /usr/share/php/Symfony/Component/Console/Command/Command.php(255): Composer\Command\InstallCommand->execute()
> #6 /usr/share/php/Symfony/Component/Console/Application.php(934): Symfony\Component\Console\Command\Command->run()
> #7 /usr/share/php/Symfony/Component/Console/Application.php(273): Symfony\Component\Console\Application->doRunCommand()
> #8 /usr/share in /mnt/data/workspace/drupal_d9/vendor/wikimedia/composer-merge-plugin/src/MergePlugin.php on line 323
> …
> ```
> (Only the first time, the next `composer install` don't trigger the error.)

_Originally posted by @prudloff-insite in https://github.com/wikimedia/composer-merge-plugin/pull/189#issuecomment-720661797_

Reply:
@mcaskill they just substitute the `composer.json` extension with `.lock` in Composer 1.10, so I would imagine it's just a case of replicating that functionality if Composer < v2.

https://github.com/composer/composer/blob/1.10/src/Composer/Factory.php#L387-L395

Reply:
I think I'll just limit the lock file backup feature to Composer 2.